src/figuresClasses.ts

```diff
@@ -1,11 +1,81 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sides = [a, b, c].sort((a1, a2) => a1 - a2);
+
+    if (a <= 0 || b <= 0 || c <= 0 || sides[2] >= sides[0] + sides[1]) {
+      throw new Error(`Sides ${a}, ${b} and ${c} are wrong`);
+    }
+  }
+
+  getArea(): number {
+    const semiPerimeter = (this.a + this.b + this.c) / 2;
+
+    const area = Math.sqrt(
+      semiPerimeter *
+        (semiPerimeter - this.a) *
+        (semiPerimeter - this.b) *
+        (semiPerimeter - this.c),
+    );
+
+    return Math.floor(area * 100) / 100;
+  }
+}
 
-export class Circle implements Figure {}
+export class Circle implements Figure {
+  public shape: Shape = 'circle';
 
-export class Rectangle implements Figure {}
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error(`Radius ${radius} is wrong`);
+    }
+  }
+
+  getArea(): number {
+    const area = this.radius * this.radius * Math.PI;
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  public shape: Shape = 'rectangle';
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+  ) {
+    if (a <= 0 || b <= 0) {
+      throw new Error(`Sides ${a} and ${b} are wrong`);
+    }
+  }
+
+  getArea(): number {
+    const area = this.a * this.b;
+
+    return Math.floor(area * 100) / 100;
```

Comment:
you doing this `Math.floor` at least three times, so it would be great to move it to a separate function